src/simulation.F90

```diff
@@ -503,6 +506,12 @@ function openmc_simulation_init() result(err) bind(C)
       else
         current_batch = restart_batch - n_realizations*n_procs
       end if
+      ! If simulation restarted from inactive batch, decrement current_batch
+      ! by one to replay an additional batch so that keff is set before resuming
+      ! simulation
+      if (restart_batch <= n_inactive) then
+        current_batch = current_batch - 1
```

Comment:
Maybe we should just have this check when a statepoint is loaded and set keff then?